extensions/samples/electron-webpack/src/main.ts

```diff
@@ -1,92 +1,89 @@
-      scopes: [
-        'user.read',
-        'people.read',
-        'user.readbasic.all',
-        'contacts.read',
-        'presence.read.all',
-        'presence.read',
-        'user.read.all',
-        'calendars.read',
-        'Sites.Read.All',
-        'Sites.ReadWrite.All',
-      ],
-      cachePlugin: new PersistenceCachePlugin(persistence),
-    };
+    static main(): void {
+        Main.application = app;
+        Main.application.on("window-all-closed", Main.onWindowAllClosed);
+        Main.application.on("ready", Main.onReady);
+        Main.authConfig = authConfig;
+    }
 
-    ElectronAuthenticator.initialize(config);
+    private static onWindowAllClosed(): void {
+        // Windows and Linux will quit the application when all windows are closed
+        if (process.platform !== "darwin") {
+            Main.application.quit();
+        }
+    }
 
-    // and load the index.html of the app.
-    mainWindow.loadFile(path.join(__dirname, "../index.html"));
+    private static async onReady(): Promise<void> {
+        try {
+            Main.createMainWindow();
+            Main.mainWindow.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
+            Main.mainWindow.on("closed", Main.onClose);
+            Main.authProvider = await AuthProvider.build(Main.authConfig);
+            Main.registerSubscriptions();
+        } catch (error) {
+            console.log(error);
+        }
+    }
 
-    // Open the DevTools.
-    mainWindow.webContents.openDevTools();
-  });
-}
+    private static createMainWindow(): void {
+        this.mainWindow = new BrowserWindow({
+            width: 800,
+            height: 600,
+            webPreferences: {
+                preload: MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY,
+                nodeIntegration: false,
+            },
+        });
+    }
+
+    private static onClose(): void {
+        Main.mainWindow = null;
+    }
 
-// This method will be called when Electron has finished
-// initialization and is ready to create browser windows.
-// Some APIs can only be used after this event occurs.
-app.on("ready", () => {
-  createWindow();
+    private static publish(message: string, payload: any): void {
+        Main.mainWindow.webContents.send(message, payload);
+    }
 
-  app.on("activate", function () {
-    // On macOS it's common to re-create a window in the app when the
-    // dock icon is clicked and there are no other windows open.
-    if (BrowserWindow.getAllWindows().length === 0) createWindow();
-  });
-});
+    private static async loadBaseUI(): Promise<void> {
+        await Main.mainWindow.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
+    }
+
+    private static async login(): Promise<void> {
+        try {
+            const account = await Main.authProvider.login();
+            if (account) {
+                await Main.loadBaseUI();
+                Main.publish("SHOW_WELCOME_MESSAGE", account);
+                Main.mainWindow.minimize();
+                Main.mainWindow.focus();
```

Comment:
I'm not seeing `.focus()` being defined anywyhere -do you actually need this?

Reply:
After acquiring the token interactively, it shows the Electron window in front of the browser screen. The `show()` API wasn't working when I tested it.

Reply:
Oh my bad this is a BrowserWindow API. Still curious though why `.show()` is not working, since we were using it in other samples -can you try this again? Also, why are we calling `mainWindow.minimize()`?

Reply:
@derisen  It's unstable on [Windows](https://github.com/electron/electron/issues/2867). I changed the workaround to use  the `setAlwaysOnTop` API to force the screen to be in the front.